.github/workflows/build_docker_image.yml

```diff
@@ -16,76 +16,55 @@ jobs:
+      - name: Docker login to GitHub
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u "${{ github.actor }}" --password-stdin
 
-  publish:
-    name: Publish
-    runs-on: ubuntu-latest
-    needs: build
-    env:
-      DOCKER_CLI_EXPERIMENTAL: enabled
-    steps:
-      - name: Get current branch
+      - name: Prepare available platforms build
+        env: # Here you need to test on what platform your docker image can be build. Important one is linux/arm/v7, linux/arm64 and linux/amd64
+          requested_platforms: "linux/amd64,linux/arm64,linux/ppc64le,linux/s390x,linux/386,linux/arm/v7,linux/arm/v6"
         run: |
-          BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/} | sed 's/\//_/g')
-          if [ "${BRANCH_NAME}" = "master" ]; then
-            BRANCH_NAME="latest"
+          # If you use the `requested_platforms` env var, then parse it.
+          if [ -z "${requested_platforms}" ]; then
+            # Transform env var into bash array to calculate arrays intersect. That allow us to know the plateform that can work with our image who can be build on current github docker buildx.
+            IFS=',' read -r -a requested_platforms <<< "${requested_platforms}"
+            IFS=',' read -r -a available_platforms <<< "${{ steps.buildx.outputs.platforms }}"
+            # Only got the intersect of two arrays
+            available_platforms=$(comm -12 <(printf '%s\n' "${requested_platforms[@]}" | LC_ALL=C sort) <(printf '%s\n' "${available_platforms[@]}" | LC_ALL=C sort))
+            # Just format the output for the docker commands
+            requested_platforms="${requested_platforms//'
+          '/,}"
+            available_platforms="${available_platforms//'
+          '/,}"
+          else
+            available_platforms="${{ steps.buildx.outputs.platforms }}"
           fi
-          echo "::set-env name=BRANCH_NAME::${BRANCH_NAME}"
-      - run: |
-          IMAGE_ID=billzimmerman/raspap
-                  
-          docker manifest create ${IMAGE_ID}:${{ env.BRANCH_NAME }} \
-                                 ${IMAGE_ID}:${{ env.BRANCH_NAME }}-amd64 \
-                                 ${IMAGE_ID}:${{ env.BRANCH_NAME }}-arm64 \
-                                 ${IMAGE_ID}:${{ env.BRANCH_NAME }}-armv7 \
-                                 ${IMAGE_ID}:${{ env.BRANCH_NAME }}-386
 
-          docker manifest push ${IMAGE_ID}:${{ env.BRANCH_NAME }}
+          echo "available_platforms=$available_platforms"
+          
+          # Save Available platforms
+          echo "available_platforms=${available_platforms}" >> $GITHUB_ENV
+      - name: Docker build & push
+        run: |
+          docker buildx build \
+          --output type=image,push=true \
+          --platform ${{ env.available_platforms }} \
+          --tag ghcr.io/${{ github.repository }}:latest \
+          .
```

Comment:
```suggestion
      # Use cache image for quicker build time.
      - name: Cache Docker layers
        uses: actions/cache@v2
        with:
          path: /tmp/.buildx-cache
          key: ${{ runner.os }}-buildx-${{ github.sha }}
          restore-keys: |
            ${{ runner.os }}-buildx-
            
      # Use official buildx GitHub Action
      - name: Docker build & push
        uses: docker/build-push-action@v2
        with:
          context: .
          push: true
          cache-from: type=local,src=/tmp/.buildx-cache
          cache-to: type=local,dest=/tmp/.buildx-cache-new
          tags: ghcr.io/${{ github.repository }}:latest
          
      # Save new cache
      - name: Move cache
        run: |
          rm -rf /tmp/.buildx-cache
          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
```

Reply:
Yeah, then we can leverage buildx cache across builds + any updates that come with this workflow